File: CPP/BasicForProgramming/Operators.cpp
#include <iostream>

using namespace std;
int a = 9;

int main() {
	/*
		Operators
			- Airthmetic
			- logical
			- bitwise *
			- relational
			- Assignment
			- others
	*/

	// Airthmetic
	int a, b;  cin >> a >> b;
	cout << a + b << endl;
	cout << a - b << endl;
	cout << a * b << endl;
	cout << a / b << endl;
	cout << a % b << endl;

	// increment, decrement by 1
	cout << a++ << " " << b++ << endl; //printed first then increase
	cout << a-- << " " << b-- << endl;
	cout << ++a << " " << ++b << endl; //incremented first then printed
	cout << --a << " " << --b << endl;

	// Relational  ->  returns boolean
	cout << (a > b) << endl;
	cout << (a < b) << endl;
	cout << (a == b) << endl;
	cout << (a != b) << endl;
	cout << (a >= b) << endl;
	cout << (a <= b) << endl;

	// Assignment -> assigning from left to right
	cout << (a = b) << endl;  // prints the value of right after assigning.
	cout << (a += b) << endl;
	cout << (a -= 1) << endl;
	cout << (a *= b) << endl;
	cout << (a /= b) << endl;
	cout << (a %= b) << endl;

	// Logical -> returns boolean
	cout << (a && a) << endl;  // truthy list in c++ contains all +ive and -ive exxcept 0
	cout << (-4 && 8) << endl;
	cout << (-6 && -8) << endl;
	cout << (a && b) << endl;
	cout << (a && 0) << endl;
	cout << (a || b) << endl;
	cout << (0 || 0) << endl;
	cout << !a << endl; // truthy value turns to falsy value -> 0, but not stored
	cout << (a = !a) << " " << a << endl;

	// Other
	cout << sizeof(int) << endl;  //gives size of anything passed
	cout << &a << endl;  // gives adress of the variable storage
	cout << a << endl;   // gives the local scope variable
	cout << :: a << endl;// gives the global scope variable

	return 0;
}

File: JAVA/BasicForProgramming/Operators.java
import java.util.*;
import java.io.*;

public class Operators {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		PrintStream out = new PrintStream(System.out);

		int a = in.nextInt(), b = in.nextInt();
		/*
			Operators
				- Airthmetic
				- logical
				- bitwise *
				- relational
				- Assignment
				- others
		*/

		// Airthematic
		out.println(a + b);
		out.println(a - b);
		out.println(a * b);
		out.println(a / b);
		out.println(a % b);

		// increment Decrement
		out.println(a++);  // print then increment
		out.println(a--);
		out.println(--a);  // decrease then print
		out.println(++a);

		// Relational
		out.println(a < b);
		out.println(a > b);
		out.println(a == b);
		out.println(a != b);

		// Logical
		/*out.println(a && b);*/  // in java there is no truthy and falsy list
		/*out.println((boolean)a && (boolean)b);*/ // but only true and false identity
		// int cannot be converted to boolean
		out.println(true && true);
		out.println(true && false);
		out.println(false || false);
		out.println(!true);

		// Assignment
		out.println(a = b);  // assigned then printed
		out.println(a += b);
		out.println(a -= 1);
		out.println(a *= b);
		out.println(a /= b);
		out.println(a %= b);
	}
}